common/git_worker.py

```diff
@@ -272,3 +275,59 @@ def get_head_revision(repo_dir):
         """
 
         return str(git.Repo(str(repo_dir)).head.commit)
+
+    @staticmethod
+    def get_last_committer_of_file(repo, file_path):
+        """
+            Get e-mail of last committer in chosen file
+
+            :param repo: path to a repository
+            :type repo: git.Git
+            :param file_path: path to a file from repo
+            :return file_path: pathlib.Path
+
+            :return file path, email of last committer
+            :rtype Tuple | None
+        """
+
+        if not file_path.is_dir():
+            rel_file_path = str(file_path.relative_to(repo.working_dir))
+            committer_email = repo.log('--format=%ae', '-1', rel_file_path)
+            return rel_file_path, committer_email
+        return None
+
+    @staticmethod
+    def get_files_owners(repos_dir, repo_states):
+        """
+            Get last committer e-mail of each file
+            of repositories from repo_states.json file
+
+            :param repos_dir: Root path to repositories
+            :type repos_dir: pathlib.Path
+            :param repo_states: List of repositories' names
+            :type repo_states: List
+
+            :return: ex: {<file_path>: <last_committer_email>, ...}
+            :rtype: Dict
+        """
+
+        repo_files = {}
+        max_workers = multiprocessing.cpu_count() * 2
+        with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
+            for repo_name in repo_states:
+                repo_path = repos_dir / repo_name
+                repo = git.Git(str(repo_path))
+                future_appends = {
+                    executor.submit(ProductState.get_last_committer_of_file, repo, file_path):
+                    file_path for file_path in repo_path.rglob('*')
+                    if '.git' not in str(file_path) and file_path.is_file()
```

Comment:
'.git' is the directory, probably file_path.is_file() is enough

Reply:
we also may have .gitignore or .gitattribute files